ohsome_quality_api/indicators/building_comparison/indicator.py

```diff
@@ -105,58 +102,52 @@ def calculate(self) -> None:
         edge_cases = [self.check_major_edge_cases(k) for k in self.data_ref.keys()]
         if all(edge_cases):
             self.result.description += (
-                " None of the reference datasets covers the area-of-interest."
+                " None of the reference datasets covers your area-of-interest."
             )
             return
-        self.result.description = ""
+        result_description = ""
         for key in self.data_ref.keys():
-            # if None in (self.ratio[key], self.area_cov[key], self.data_ref[key]):
-            if self.warnings[key] != "":
-                self.result.description += self.warnings[key] + "\n"
+            # if major edge case present add to description
+            # and continue with next dataset
+            edge_case = self.check_major_edge_cases(key)
+            if edge_case != "":
+                result_description += edge_case + " "
                 continue
 
-            self.warnings[key] += self.check_minor_edge_cases(key)
+            edge_case = self.check_minor_edge_cases(key)
             # TODO: check for None explicitly?
             # TODO: add warning for user, that no buildings are present?
-            try:
-                self.ratio[key] = self.area_osm[key] / self.area_ref[key]
-
-                template = Template(self.metadata.result_description)
-                self.warnings[key] += template.substitute(
-                    ratio=round(self.ratio[key] * 100, 2),
-                    coverage=round(self.area_cov[key] * 100, 2),
-                    dataset=self.data_ref[key]["name"],
-                )
-            except ZeroDivisionError:
-                self.ratio[key] = None
-                self.warnings[key] += (
-                    f"Warning: Reference dataset {self.data_ref[key]['name']} covers "
-                    f"AoI with {round(self.area_cov[key] * 100, 2)}%, but has no "
-                    "building area. No quality estimation with reference is possible. "
-                )
-            self.result.description += self.warnings[key] + "\n"
+            # ZeroDivisionError can not occur because of `check_edge_cases()`
```

Comment:
Are you sure about this? If I remember correctly we had some Issues with ZeroDivisionErrors, as the `check_edge_cases()` checks for coverage, but the division is done with the building area. So, depending on the AoI, it may be covered by the referenz dataset (so no edge case present) but no buildings are present so the building area is zero, resulting in ZeroDivisionError. Same would apply in the road comparison indicator

Reply:
Yes, I check explicitly for `None` in `check_major_edge_cases`. If major edge case is present above code will not be executed.